### PR TITLE
[CORE-1723] Add/update tests for project defaults

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

**Comment by robert-uhl** [comment]
I did spend some time trying to abstract out some of the test patterns common to defaults, but ultimately realized that the abstracted version was brittle and complex, so I ended up just going with straightforward imperative tests.

**Review by acohen4 — APPROVED** [review]
LGTM. One nit. Also, do we have Project Default tests for a project other than default?

**Comment by robert-uhl** [comment]
> Also, do we have Project Default tests for a project other than default?

Nope, just added two. Good catch, thanks.

File: src/server/pps/server/defaults_test.go [hunk 1]
```diff
@@ -481,13 +394,132 @@ func TestAPIServer_CreatePipelineV2_reprocessSpec(t *testing.T) {
 func TestAPIServer_GetProjectDefaults(t *testing.T) {
 	ctx := pctx.TestContext(t)
 	env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t).PachConfigOption)
+
 	_, err := env.PPSServer.GetProjectDefaults(ctx, &pps.GetProjectDefaultsRequest{})
 	require.NoError(t, err, "empty request must not be an error")
+
```
[the remark this answers is not in the document]
Comment (robert-uhl) [reply src/server/pps/server/defaults_test.go:400]: Was the intention then that a function should be short & clear, or that newlines be avoided? I can definitely see an argument that if one wants to introduce some structure into code, then the right form for that structure to take is another function. OTOH, taking that too far can lead to premature abstraction, as well as greatly increased vertical space due to function signatures and opening/closing braces. Using whitespace to separate subfunction stanzas when it doesn’t make sense to extract those seems useful.

I think that this applies even more in tests, which tend to be of the form “perform an operation, then perform a collection of tests on the results.” The operation and the checks change from stanza to stanza, hampering function abstraction. But deleting whitespace just leaves an undifferentiated wall of ASCII. OTOH, using vertical whitespace to separate each operation/validation sequence provides a bit of visual structure.

**Pull request merged**